File: DSA/Java/src/DSA/Searching/LinearSearch.java
package DSA.Searching;

public class LinearSearch {
    public static void main(String[] args) {
        int[] arr = {2, 0, -1, 42, 34, 13};
        int target = 13;
        int result = linearSearch(arr, target);
        System.out.println(result);
    }

    private static int linearSearch(int[] arr, int target) {
        for (int index = 0; index < arr.length; index++) {
            if(target == arr[index]){
                return index;
            }
        }
        return -1;
    }
}

File: DSA/100. Old Data/Searches/LinearSearch.cpp
#include <iostream>
using namespace std;

int linearSearch(int arr[], int length, int element){
   int index = -1;
   for(int i = 0; i<length; i++){
      if(arr[i] == element){
         index = i;
         return index;
      }
   }
   return index;
}

int main(){
   int arr[] = {5, 7, -2, 10, 22, -5, 0, 6, 45, 1};

   // Checking whether 6 is present in the array or not -
   int arrLength = sizeof(arr) / sizeof(int);
   int answer = linearSearch(arr, arrLength, -6);
   cout << answer << endl;
   return 0;
}